htmresearch/encoders/cio_encoder.py

```diff
@@ -102,6 +117,15 @@ def encode(self, text):
                "the corpus.".format(text))
       encoding = self._subEncoding(text)
 
+    if self.retinaScaling != 1:
+      encoding["fingerprint"]["positions"] = self.scaleEncoding(
+        encoding["fingerprint"]["positions"], self.retinaScaling, dimensions=2.0)
```

Comment:
You're right, it's better to always use 1D and just modify the scale accordingly; e.g. instead of scaling 128x128 to 64x64 w/ scale=0.5 and dimensions=2, we'll just do scale=0.25.